File: OpenSilver.Samples.TelerikUI/OpenSilver.Samples.TelerikUI/Samples/Controls/RadOutlookBar/MenuItems.cs
using System.Collections.ObjectModel;

namespace OpenSilver.Samples.TelerikUI.RadOutlookBar
{
    public class MenuItemBase
    {
        public string Header { get; set; }
        public string Content { get; set; }
        public string IconSource { get; set; }
        public string IconSourceSmall { get; set; }

        public override string ToString()
        {
            return string.Empty;
        }
    }

    public class MailMenuItem : MenuItemBase
    {
        public ObservableCollection<MailDirectoryItem> MailDirectories { get; set; }
        public MailMenuItem()
        {
            MailDirectories = new ObservableCollection<MailDirectoryItem>();
        }
    }

    public class MailDirectoryItem
    {
        public string Header { get; set; }
        public string IconSource { get; set; }
        public ObservableCollection<MailDirectoryItem> Children { get; set; }

        public MailDirectoryItem()
        {
            Children = new ObservableCollection<MailDirectoryItem>();
        }
    }

    public class CalendarMenuItem : MenuItemBase
    {
    }

    public class ContactsMenuItem : MenuItemBase
    {
        public ObservableCollection<Person> ContactsList { get; set; }
    }

    public class Person
    {
        public string Name { get; set; }
        public string IconSource { get; set; }
    }
}

File: OpenSilver.Samples.TelerikUI/OpenSilver.Samples.TelerikUI/Samples/Controls/RadOutlookBar/MenuItems.vb
Imports System.Collections.ObjectModel

Namespace OpenSilver.Samples.TelerikUI.RadOutlookBar
    Public Class MenuItemBase
        Public Property Header As String
        Public Property Content As String
        Public Property IconSource As String
        Public Property IconSourceSmall As String

        Public Overrides Function ToString() As String
            Return String.Empty
        End Function
    End Class

    Public Class MailMenuItem
        Inherits MenuItemBase
        Public Property MailDirectories As ObservableCollection(Of MailDirectoryItem)
        Public Sub New()
            MailDirectories = New ObservableCollection(Of MailDirectoryItem)()
        End Sub
    End Class

    Public Class MailDirectoryItem
        Public Property Header As String
        Public Property IconSource As String
        Public Property Children As ObservableCollection(Of MailDirectoryItem)

        Public Sub New()
            Children = New ObservableCollection(Of MailDirectoryItem)()
        End Sub
    End Class

    Public Class CalendarMenuItem
        Inherits MenuItemBase
    End Class

    Public Class ContactsMenuItem
        Inherits MenuItemBase
        Public Property ContactsList As ObservableCollection(Of Person)
    End Class

    Public Class Person
        Public Property Name As String
        Public Property IconSource As String
    End Class
End Namespace
